unstable/wlr-exec-secure-unstable-v1.xml

```diff
@@ -0,0 +1,139 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="wlr_exec_secure_unstable_v1">
+  <copyright>
+    Copyright © 2018 Drew DeVault
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice (including the next
+    paragraph) shall be included in all copies or substantial portions of the
+    Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+  </copyright>
+
+  <description summary="a protocol for executing clients with higher permissions">
+    This protocol allows unprivileged clients to ask the compositor to execute
+    a program which can access protected Wayland protocols.
+
+    The rationale for this protocol is that the execution environment of an
+    arbitrary client cannot be reliably verified by the compositor. Environment
+    variables like LD_PRELOAD may alter the runtime behavior of the client to
+    malicious ends. However, if the compositor creates the client socket and
+    exec's the client program itself, it can be sure of the client's execution
+    environment.
+
+    If a client is run with the WAYLAND_SOCKET environment variable set, it can
+    assume that the socket represents an authenticated Wayland connection and
+    it will be able to use any secure protocols it is afforded. If not, the
+    client will likely connect to the Unix socket specified by WAYLAND_DISPLAY,
+    but it will be an unauthenticated connection. If it needs authenticated
+    protocols to accomplish its basic tasks, it will likely wish to use
+    wlr_exec_secure_unstable_v1 to ask the compositor to execute itself again.
+    It's at the client's discretion to exit once the compositor runs the second
+    process or to wait for it to complete.
```

Comment:
I think it's unlikely that is how it's going to pan out.
It'll probably end up being `wayland-exec ./my-prog args`, where this hypothetical wayland-exec is just a small program that handles all of this. Or otherwise small interactive launcher programs which try to open the real program.

Reply:
I think it's more likely that if `my-prog` wants to access Wayland protocols it knows require authentication, it will naturally include a means to authenticate. This would become a basic requirement of that software.